src/Stripe.Tests.XUnit/plans/_cache.cs

```diff
@@ -23,7 +23,7 @@ public static StripePlanCreateOptions GetPlanCreateOptions(string planName = "pl
                 Amount = 1000,
                 Currency = "usd",
                 Name = Guid.NewGuid().ToString(),
-                Id = Guid.NewGuid().ToString(),
+                Id = Guid.NewGuid().ToString() + DateTime.UtcNow.Ticks,
```

Comment:
Any idea why this works? A GUID should already be universally unique (unless some crazy stubbing has been applied).

Reply:
@brandur-stripe well I think I'm just unlucky enough? I keep getting "plan already exists" over and over when running the test suite and those conflict on the random ids. But I have ran the test suites hundreds of times so maybe it's due to that?

Reply:
Just for general interest, I looked it up and .NET uses a V4 UUID generator (random).

Here's a [good writeup from a Quora answer](https://www.quora.com/Has-there-ever-been-a-UUID-collision):

> UUIDs have 122 bits of entropy so the chance of two random UUIDs colliding is about 10^-37.
>
> If you generate 2^46 UUIDs (approximately 1 petabyte of entropy) the chance of getting a collision is 1 in 50 billion.[1]  You're 170 times more likely to win the jackpot in Powerball.[2]

Meaning to say that the chances of generating a duplicate GUID (modulo a bug in the underlying software) are ~zero. I guess it's not a big deal, but I think that the required addition of extra entropy to this string probably means that we have some other bug or feature that we don't completely understand somewhere.

Reply:
I think you're right though I'm not grasping why I experience this. My ids are definitely GUIDs and I can reproduce reliably. Unless I add some logs that is, when I do the error does not happen anymore.
I guess this is the wrong fix and we should understand what's causing this issue in the first place.

Reply:
Okay, I think the issue here is that we go into `GetPlan()` and the plan does not exist yet. But when we go into `GetPlanCreateOptions()` another concurrent call to `GetPlanCreateOptions()` already initialized the options in the `Items` array. This causes another part of the code to try and create a plan a second time with the same option.

I do not understand why adding a timestamp fix those. It's likely a side-effect and not a solution per se.

I guess the right solution is to take a lock in `GetPlan()` and release it only when leaving.

Reply:
removed this attempt for now as it's the wrong approach.